src/main/java/co/rsk/federate/FederationProviderFromFederatorSupport.java

```diff
@@ -93,7 +100,7 @@ public Optional<Federation> getRetiringFederation() {
         Integer federationSize = federatorSupport.getRetiringFederationSize();
         Optional<Address> optionalRetiringFederationAddress = getRetiringFederationAddress();
 
-        if (federationSize == -1 || !optionalRetiringFederationAddress.isPresent()) {
+        if (federationSize == FEDERATION_NON_EXISTENT.getCode() || optionalRetiringFederationAddress.isEmpty()) {
```

Comment:
I know this already existed here, but probably only one of the checks is enough right? I'd just stick to checking the federation size

Reply:
Sounds good I have removed the extra check. I placed the call after the check. I also modifed the corresponding tests. I deleted one since it didn't make sense to have it anymore IMO. Let me know if that is fine.